documentation/components.md

```diff
@@ -11,75 +11,137 @@ const Parent = () => (
   </section>
 );
 
-const Label = ({ greeting, children }) => (
+const Label = (props) => (
   <>
-    <div>{greeting}</div>
-    {children}
+    <div>{props.greeting}</div>
+    {props.children}
   </>
 );
 ```
 
-Since all nodes from JSX are actual DOM nodes the only responsibility of top level Templates/Components is appending to the DOM.
+Since all nodes from JSX are actual DOM nodes, the only responsibility of top level Templates/Components is appending to the DOM.
 
-Components also support dynamic bindings which allow you to pass values that will change. However you need to be careful to access your props inside bindings or effects if you want them to track change.
+## Props
+
+Much like React, Vue, Angular, and other frameworks, you can define properties on your components which allow a parent component to pass data to a child component. Here a parent is passing the string "Hello" to the `Label` component via a `greeting` property.
 
 ```jsx
-// Name will never update as it is destructured outside
-const StaticComponent = ({ name }) => <div>{ name }</div>
+const Parent = () => (
+  <section>
+    <Label greeting="Hello">
+      <div>John</div>
+    </Label>
+  </section>
+);
+```
 
-// Updates like you'd expect
-const DynamicComponent = props => <div>{ props.name }</div>
+In the above example, the value set on `greeting` is static, but we can also set dynamic values. For example:
 
-// Update on state.name change
-<DynamicComponent name={ state.name }/>
+```jsx
+const Parent = () => (
+  let value = Math.random().toString();
 
-// will not update on name change and pass by value
-const { name } = state;
-<DynamicComponent name={ name }/>
+  <section>
+    <Label greeting={value}>
+      <div>John</div>
+    </Label>
+  </section>
+);
+```
+
+Components can access properties passed to them via a `props` argument.
 
-// Still won't update even with the dynamic binding
-<StaticComponent name={ state.name }/>
+```jsx
+const Label = (props) => (
+  <>
+    <div>{props.greeting}</div>
+    {props.children}
+  </>
+);
 ```
 
-## Children
+Unlike in some other frameworks (e.g. React), you cannot use object destructuring on the `props` of a component. This is because the `props` object is, behind the scenes, a Proxy object that relies on getters to lazily retrieve values. Using object descructuring breaks the reactivity of `props`.

```

Comment:
I'm not sure how accurate this description of the `props` implementation is, but I think it paints a clearer picture than the current description in the docs.

Reply:
To be precise, props are not Proxies unless you use `mergeProps`/`splitProps`. Normally they are just plain objects with regular properties (for static props) and getters (for reactive props), but that's probably too much detail for this doc.